# **PUNTO 1**

## Librerías y módulos

In [1]:
import warnings
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
warnings.filterwarnings("ignore")
from IPython.display import display

En primer lugar, procedemos a importar la base de datos proporcionada.

In [2]:
datatrain = pd.read_csv('C:/Users/valcd/OneDrive/Escritorio/MachinelearningUN/Parcial_3/satisfaccion_de_clientes_class/satisfaccion_de_clientes_class/train.csv')
datatrain

ID  var3  var15  imp_ent_var16_ult1  imp_op_var39_comer_ult1  \
0           1     2     23                 0.0                      0.0   
1           3     2     34                 0.0                      0.0   
2           4     2     23                 0.0                      0.0   
3           8     2     37                 0.0                    195.0   
4          10     2     39                 0.0                      0.0   
...       ...   ...    ...                 ...                      ...   
76015  151829     2     48                 0.0                      0.0   
76016  151830     2     39                 0.0                      0.0   
76017  151835     2     23                 0.0                      0.0   
76018  151836     2     25                 0.0                      0.0   
76019  151838     2     46                 0.0                      0.0   

       imp_op_var39_comer_ult3  imp_op_var40_comer_ult1  \
0                          0.0                      0.0   
1                          0.0                      0.0   
2                          0.0                      0.0   
3                        195.0                      0.0   
4                          0.0                      0.0   
...                        ...                      ...   
76015                      0.0                      0.0   
76016                      0.0                      0.0   
76017                      0.0                      0.0   
76018                      0.0                      0.0   
76019                      0.0                      0.0   

       imp_op_var40_comer_ult3  imp_op_var40_efect_ult1  \
0                          0.0                      0.0   
1                          0.0                      0.0   
2                          0.0                      0.0   
3                          0.0                      0.0   
4                          0.0                      0.0   
...                        ...                      ...   
76015                      0.0                      0.0   
76016                      0.0                      0.0   
76017                      0.0                      0.0   
76018                      0.0                      0.0   
76019                      0.0                      0.0   

       imp_op_var40_efect_ult3  ...  saldo_medio_var33_hace2  \
0                          0.0  ...                      0.0   
1                          0.0  ...                      0.0   
2                          0.0  ...                      0.0   
3                          0.0  ...                      0.0   
4                          0.0  ...                      0.0   
...                        ...  ...                      ...   
76015                      0.0  ...                      0.0   
76016                      0.0  ...                      0.0   
76017                      0.0  ...                      0.0   
76018                      0.0  ...                      0.0   
76019                      0.0  ...                      0.0   

       saldo_medio_var33_hace3  saldo_medio_var33_ult1  \
0                          0.0                     0.0   
1                          0.0                     0.0   
2                          0.0                     0.0   
3                          0.0                     0.0   
4                          0.0                     0.0   
...                        ...                     ...   
76015                      0.0                     0.0   
76016                      0.0                     0.0   
76017                      0.0                     0.0   
76018                      0.0                     0.0   
76019                      0.0                     0.0   

       saldo_medio_var33_ult3  saldo_medio_var44_hace2  \
0                         0.0                      0.0   
1                         0.0                      0.0   
2                         0.0                      0

In [4]:
datatrain.shape

(76020, 371)

La satisfacción del cliente es clave para el éxito organizacional, especialmente en una entidad financiera donde los clientes insatisfechos suelen abandonar la empresa sin previo aviso. Identificar a estos clientes de manera temprana permitiría implementar estrategias preventivas para mejorar su experiencia y fomentar su fidelización.

El conjunto de datos **Satisfacción de Clientes** contiene **76,020 observaciones** anonimizadas con **371 variables numéricas** y una columna `TARGET`, que clasifica a los clientes en satisfechos (`0`) o insatisfechos (`1`). Este análisis busca desarrollar modelos predictivos para anticipar la insatisfacción y mejorar la relación con los clientes.